### DISCUSSION TITLE
.raw file "Image too big for processing"

### DISCUSSION THREAD

**Discussion by AaronMuscarella** [ImageMagick/ImageMagick#6922] [Help]
### ImageMagick version

7.1.1-22

### Operating system, version and so on

Windows 10

### Description

I have a large .raw file (20gb) that I'm trying to convert to TIFF64 using 
magick.exe convert input.raw TIFF64:output.tif
but keep getting "Image too big for processing" error. I'm not sure what to do. Any suggestions? I'd be fine splitting it up but that gives the same error.
thanks

**Comment by snibgo** [comment]
Please show the complete error message.

Is your raw file from a digital camera, or what?

What is the text output from `magick input.raw -verbose info:`?

**Comment by AaronMuscarella** [comment]
error message:
convert: Image too big for processing `input.raw' @ error/dng.c/ReadDNGImage/547.

the file is from an electron microscope

text output:
magick: Image too big for processing `K:\Data\9_Karin\5-Atlas\048_5xFAD_MtOGG_Stiched.raw' @ error/dng.c/ReadDNGImage/547.

thanks

**Comment by snibgo** [comment]
That error has been thrown by libraw_open_file(), which is in the libraw library. It seems it is hitting ...
```
#define LIBRAW_MAX_ALLOC_MB 2048L
```
... in `libraw_const.h`. If your raw file is 20GB, there is no chance the pixels will fit in 2048MB, ie 2GB. Try setting that number very much higher, say 40960, then rebuilding libraw and ImageMagick.

I assume your computer has loads of memory.

If that doesn't help, I suggest you refer the question to the libraw people.

**Reply by dlemstra** [reply]
It seems that we can override this value with the `max_raw_memory_mb` property. We should probably expose this option as a define to allow reading that file.
